pkg/kubectl/resource/builder.go

```diff
@@ -757,3 +774,17 @@ func HasNames(args []string) (bool, error) {
 	}
 	return hasCombinedTypes || len(args) > 1, nil
 }
+
+func (b *Builder) validateNamespace(c RESTClient, namespace string) error {
+	if !b.validateNamespaces {
+		return nil
+	}
+	if namespace == api.NamespaceDefault || namespace == api.NamespaceAll {
+		return nil
+	}
+	err := c.Get().AbsPath("api", "v1", "namespaces", namespace).Do().Error()
```

Comment:
I would only error on a 404 (and not a generic 404, a specific 404 from the server).  That would be `if err != nil && errors.IsNotFound(err) && !errors.<SOMETHING_SPECIFIC>`.  I.e. if the server returns 404 without returning an actual error, we shouldn't exit out here.  I wouldn't want this to mask other errors.

Also, can't we do this on the server and communicate it cleanly to clients.